tests/unit/amo/components/TestAddon.js

```diff
@@ -255,6 +255,28 @@ describe('Addon', () => {
     expect(root.find(NotFound)).toHaveLength(1);
   });
 
+  it('renders NotFound page for unauthorized add-on - 401 error', () => {
+    const id = 'error-handler-id';
+    const { store } = createStore();
+
+    const error = createApiError({
+      response: { status: 401 },
+      apiURL: 'https://some/api/endpoint',
+      jsonResponse: { message: 'Authentication credentials were not provided.' },
+    });
+    store.dispatch(setError({ id, error }));
+    const capturedError = store.getState().errors[id];
+    expect(capturedError).toBeTruthy();
```

Comment:
Do we actually need this? It does not seem to cover any code that you have changed or am I missing something?